File: Java/Tests/jdi-uitest-tutorialtests/src/main/java/com/epam/jdi/uitests/testing/career/page_objects/site/pages/JobListingPage.java
package com.epam.jdi.uitests.testing.career.page_objects.site.pages;

import com.epam.commons.map.MapArray;
import com.epam.jdi.uitests.testing.career.page_objects.enums.JobListHeaders;
import com.epam.jdi.uitests.web.selenium.elements.complex.table.Table;
import com.epam.jdi.uitests.web.selenium.elements.complex.table.interfaces.ICell;
import com.epam.jdi.uitests.web.selenium.elements.complex.table.interfaces.ITable;
import com.epam.jdi.uitests.web.selenium.elements.composite.WebPage;
import org.openqa.selenium.By;
import org.openqa.selenium.support.FindBy;
import com.epam.jdi.uitests.web.selenium.elements.complex.Dropdown;
import com.epam.jdi.uitests.core.interfaces.complex.IDropDown;
import static com.epam.jdi.uitests.testing.career.page_objects.enums.JobListHeaders.APPLY;
import static com.epam.jdi.uitests.testing.career.page_objects.enums.JobListHeaders.JOB_NAME;
import static com.epam.jdi.uitests.web.selenium.elements.complex.table.Column.column;

/**
 * Created by Roman_Iovlev on 10/22/2015.
 */
public class JobListingPage extends WebPage {

    @FindBy (className = "selected-params")
    public IDropDown<Enum> categories = new Dropdown<>(By.xpath("//*[contains(@class,'selected-params')]"),
            By.xpath("//*[*[contains(@class,'selected-params')]]//li"));

    @FindBy(name = "location_select")
    public IDropDown<Enum> location = new Dropdown<>(By.xpath("//*[*[contains(@id,'select-box-location-select-container')]]//*[@class='arrow']"),
            By.xpath("//*[contains(@id, 'location-select')]//li"));

    @FindBy(className = "search-result-list")
    public ITable jobsList = new Table(null,
            By.xpath(".//li[%s]//div"),
            By.xpath(".//li//div[%s]"))
            .hasColumnHeaders(JobListHeaders.class);

    public void getJobRowByName(String jobName) {
        MapArray<String, ICell> row = jobsList.row(jobName, column(JOB_NAME.toString()));
        row.get(APPLY.toString()).select();
    }
}

File: C# .Net/JDI UI Framework/Tests/Scenarios/Page Objects/Pages/JobListingPage.cs
using System;
using Epam.JDI.Core.Interfaces.Complex;
using Epam.JDI.Web.Attributes;
using Epam.JDI.Web.Selenium.Elements.Complex;
using Epam.JDI.Web.Selenium.Elements.Complex.table;
using Epam.JDI.Web.Selenium.Elements.Complex.table.interfaces;
using Epam.JDI.Web.Selenium.Elements.Composite;
using Epam.Tests.Scenarios.Enums;
using OpenQA.Selenium;
using static Epam.JDI.Web.Selenium.Elements.Complex.table.Column;
using static Epam.Tests.Scenarios.Enums.JobListHeaders;

namespace Epam.Tests.Scenarios.Page_Objects.Pages
{
    public class JobListingPage : WebPage
    {
        [FindBy(ClassName = "selected-params")]
            public IDropDown<Enum> categories = new Dropdown<Enum>(By.XPath("//*[contains(@class,'selected-params')]"),
            By.XPath("//*[*[contains(@class,'selected-params')]]//li"));

        [FindBy(Name = "location_select")]
            public IDropDown<Enum> location = new Dropdown<Enum>(By.XPath("//*[*[contains(@id,'select-box-location-select-container')]]//*[@class='arrow']"),
            By.XPath("//*[contains(@id, 'location-select')]//li"));


        [FindBy(ClassName = "search-result-list")]
        public ITable JobsList = new Table(null,
                By.XPath(".//li[{0}]//div"),
                By.XPath(".//li//div[{0}]"))
                .HasColumnHeaders(typeof(JobListHeaders));

        public void GetJobRowByName(string jobName)
        {
            var row = JobsList.Row(jobName, column(JobName));
            row[Apply].Select();
        }
    }
}
